pulp/tests/test_gurobipy_env.py

```diff
@@ -14,6 +14,15 @@ def check_dummy_env():
         pass
 
 
+def is_single_use_license() -> bool:
```

Comment:
the `SkipTest` decorators below failed to type check. I was deciding between this and just unilateral skip with `@unittest.skip`. Happy to go the other way if you want.

Reply:
I don't really understand the difference between `unittest.SkipTest` and `unittest.skip`. Maybe @torressa knows ?

Reply:
Is there a way to check for the single use license inside the `is_single_use_license` function and return true/false?

Reply:
With `gurobipy` one could do:

```python
def is_single_use_license() -> bool:
    try:
        env1 = gp.Env()
        env2 = gp.Env()
        env2.dispose()
        env1.dispose()
    except gp.GurobiError as ge:
        if ge.errno == gp.GRB.Error.NO_LICENSE:
            print("single use license")
            print(f"Error code {ge.errno}: {ge}")
            return True
    return False
```

This is enough to trigger an error in a single-use case.